Content.Shared/_CD/NanoChat/SharedNanoChatSystem.cs

```diff
@@ -314,6 +315,44 @@ public bool EnsureRecipientExists(Entity<NanoChatCardComponent?> card,
         Dirty(card);
         return true;
     }
+    // Moffstation - Begin - Added the ability for cards sharing the same Nanochat Number to syncronise outgoing messages
+    /// <summary>
+    ///     Syncronises messages across every card with the same number, ensuring
+    ///     message echo fan out and in.
+    /// </summary>
+    public void SyncMessagesForCard(Entity<NanoChatCardComponent?> card)
+    {
+        if (!Resolve(card, ref card.Comp) || !card.Comp.Number.HasValue)
+        {
+            return;
+        }
 
+        var messagesMerged = new Dictionary<uint, List<NanoChatMessage>>();
+        var cards =
+            EntityQueryEnumerator<NanoChatCardComponent>(); //added so that this methode only has to be called once :3
```

Comment:
```suggestion
        var cards = EntityQueryEnumerator<NanoChatCardComponent>();
```
:3 eliminated to shorten the line